quickshell/Services/WeatherService.qml

```diff
@@ -149,6 +150,127 @@ Singleton {
+                                       "12": "",
+                                       "13": "",
+                                       "14": "",
+                                       "15": "",
+                                       "16": "",
+                                       "17": "",
+                                       "18": "",
+                                       "19": "",
+                                       "20": "",
+                                       "21": "",
+                                       "22": "",
+                                       "23": "",
+                                       "24": "",
+                                       "25": "",
+                                       "26": "",
+                                       "27": "",
+                                   })
+
+    property var moonMaterialIcons: ({
+                                        "0":  "󰽤",
+                                        "1":  "󰽧",
+                                        "2":  "󰽡",
+                                        "3":  "󰽦",
+                                        "4":  "󰽢",
+                                        "5":  "󰽨",
+                                        "6":  "󰽣",
+                                        "7":  "󰽥",
+                                    })
+
+    function getMoonPhase(date) {
+        const icons = moonWeatherIcons // more icons in this set but thinner outline than material icons
+        // const icons = moonMaterialIcons
+        const iconCount = Object.keys(icons).length
+        const moon = SunCalc.getMoonIllumination(date)
+        const index = ((Math.floor(moon.phase * iconCount + 0.5) % iconCount) + iconCount) % iconCount
+
+        return icons[index]
+    }
+
+    function getMoonAngle(date) {
+        const pos = SunCalc.getMoonPosition(date, location.latitude, location.longitude)
+        return pos.parralacticAngle
+    }
+
+    function getLocation() {
+        return location
+    }
+
+    function getSunTimes(date) {
+        // NOTE: this can take a height argument (meters above sealevel)
+        return SunCalc.getTimes(date, location.latitude, location.longitude)
+    }
+
+    function getCurrentSunTime(date) {
+        const times = getSunTimes(date)
+        const dateObj = new Date(date)
+
+        const periods = [
+            { name: "Dawn (Astronomical Twilight)", start: new Date(times.nightEnd),      end: new Date(times.nauticalDawn) },
+            { name: "Dawn (Nautical Twilight)",     start: new Date(times.nauticalDawn),  end: new Date(times.dawn) },
+            { name: "Dawn (Civil Twilight)",        start: new Date(times.dawn),          end: new Date(times.sunrise) },
+            { name: "Sunrise",                      start: new Date(times.sunrise),       end: new Date(times.sunriseEnd) },
+            { name: "Golden Hour",                  start: new Date(times.sunriseEnd),    end: new Date(times.goldenHourEnd) },
```

Comment:
Good to add I18n.tr() around all user-facing strings, that will get them parsed out for translations (I knoww that some are missed in the codebase like the other weather ones above)